nullaway/src/test/java/com/uber/nullaway/SyncLambdasTests.java

```diff
@@ -0,0 +1,163 @@
+ */
+public class SyncLambdasTests extends NullAwayTestsBase {
+
+  @Test
+  public void forEachOnMap() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import java.util.Map;",
+            "import java.util.HashMap;",
+            "import org.jspecify.annotations.Nullable;",
+            "public class Test {",
+            "    private @Nullable Map<Object, Object> target;",
+            "    private @Nullable Map<Object, Object> resolved;",
+            "    public void initialize() {",
+            "        if (this.target == null) {",
+            "            throw new IllegalArgumentException();",
+            "        }",
+            "        this.resolved = new HashMap<>();",
+            "        this.target.forEach((key, value) -> {",
+            "            // no error here as info gets propagated",
+            "            this.resolved.put(key, value);",
+            "        });",
+            "    }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void forEachOnHashMap() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import java.util.HashMap;",
+            "import org.jspecify.annotations.Nullable;",
+            "public class Test {",
+            "    private @Nullable HashMap<Object, Object> target;",
+            "    private @Nullable HashMap<Object, Object> resolved;",
+            "    public void initialize() {",
+            "        if (this.target == null) {",
+            "            throw new IllegalArgumentException();",
+            "        }",
+            "        this.resolved = new HashMap<>();",
+            "        this.target.forEach((key, value) -> {",
+            "            // no error here as info gets propagated",
+            "            this.resolved.put(key, value);",
+            "        });",
+            "    }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void otherForEach() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import java.util.HashMap;",
+            "import java.util.function.BiConsumer;",
+            "import org.jspecify.annotations.Nullable;",
+            "public class Test {",
+            "    private @Nullable MyMap<Object, Object> target;",
+            "    private @Nullable MyMap<Object, Object> resolved;",
+            "    static class MyMap<K,V> {",
+            "        public void forEach(BiConsumer<Object, Object> consumer) {}",
+            "        public void put(Object key, Object value) {}",
+            "    }",
+            "    public void initialize() {",
+            "        if (this.target == null) {",
+            "            throw new IllegalArgumentException();",
+            "        }",
+            "        this.resolved = new MyMap<>();",
```

Comment:
Maybe use a type other than `MyMap` for `this.resolved` here? Got a bit confused think that was what changed the test case, but it doesn't, right? What matters is that `target` is `MyMap`.

Reply:
8a036763e0ee0239f6a8f7d6ffb1d9e1dcdb7925